File: nandincpp/Minimum Number of Swaps to Make the String Balanced.java
class Solution {
    public int minSwaps(String s) 
    {
        Stack<Character> stk = new Stack<> ();
        
        for(char c : s.toCharArray())
        {
            if(c == '[')
                stk.push(c);
            else
            {
                if(!stk.isEmpty())
                    stk.pop();
            }
        }
        
        return (stk.size()+1)/2;
    }
}

File: nandincpp/Minimum Number of Swaps to Make the String Balanced.cpp
class Solution {
public:
    int minSwaps(string s)
    {
        stack<int> stk;
        
        for(auto i : s)
        {
            if(i == '[')
                stk.push(i);
            else
            {
                if(!stk.empty())
                    stk.pop();
            }
        }
        
        return (stk.size()+1)>>1;
    }
};